src/persist-txn/src/txn_cache.rs

```diff
@@ -362,7 +358,6 @@ impl<T: Timestamp + Lattice + TotalOrder + StepForward + Codec64> TxnsCacheState
 
     /// Returns the minimum timestamp not known to be applied by this cache.
     pub fn min_unapplied_ts(&self) -> &T {
-        assert_eq!(self.only_data_id, None);
```

Comment:
Is this an important `assert`? I couldn't tell.

Reply:
Turns out I'm able to add it back because I'm no longer using this method in the same places.

Reply:
I added an inner method that is not public that I can call from other not public methods (such as `validate`).